src/plugin/base-plugin.js

```diff
@@ -123,6 +123,17 @@ export default class BasePlugin implements IPlugin {
   }
 
   /**
+   * Runs the reset logic of the plugin.
+   * plugin must implement this method.
+   * @public
+   * @abstract
+   * @returns {void}
+   */
+  reset(): void {
+    throw new PlayerError(PlayerError.TYPE.NOT_IMPLEMENTED_METHOD, 'reset()').getError();
```

Comment:
maybe a plugin has nothing to do in `reset`. better to leave it empty.

Reply:
I think it the same as 'destroy' - we need to force plugins to implement those methods